### ISSUE TITLE
object x has a Value with the wrong type

### ISSUE THREAD

**Issue by zeromancer** [lloydmeta/enumeratum#119]
I get the following compile error:
```
Error(73,45): object Example1 has a Value with the wrong type: Hello World:class java.lang.String, instead of int.
val values: IndexedSeq[BugDemonstration] = findValues
```
* I noticed that i get errors like that if i have functions inside objects that extend an Enum (like Example1)
* The more complex the code in the function is - the more likely it is to happen.
* I tried to use Strings instead of Table/Actor (of libgdx library) -> but then the error magically disappears ....
* Bizarre behaviour: if "case object Example1" is commented out -> the code compiles and launches successfully
* can this be fixed, or am I doing something wrong ?

Code to reproduce error:
```
import com.badlogic.gdx.backends.lwjgl.{LwjglApplication, LwjglApplicationConfiguration}
import com.badlogic.gdx.graphics.GL20
import com.badlogic.gdx._
import com.badlogic.gdx.scenes.scene2d._
import com.badlogic.gdx.scenes.scene2d.ui._
import com.badlogic.gdx.utils.viewport.ScreenViewport
import enumeratum.values.{IntEnum, IntEnumEntry}

import scala.collection.immutable.IndexedSeq

object Start1 extends App with ApplicationListener {

	var skin: Skin = _
	var stage: Stage = _
	var lastSelected: String = _
	var container: Table = _
	var table: Actor = _

	override def create(): Unit = {

		stage = new Stage(new ScreenViewport())
		skin = new Skin()

		// Container: List + Table
		container = new Table()
		container.setFillParent(true)
		stage.addActor(container)

		val a = BugDemonstration.Example2.create(skin)
		container.add(a)
	}

	override def resize(width: Int, height: Int): Unit = stage.getViewport.update(width, height, true)

	override def render(): Unit = {
		Gdx.gl.glClearColor(0.2f, 0.2f, 0.2f, 1)
		Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT)
		stage.act()
		stage.draw()
	}

	override def pause(): Unit = {}

	override def resume(): Unit = {}

	override def dispose(): Unit = {
		stage.dispose()
		skin.dispose()
	}

	val config = new LwjglApplicationConfiguration()
	config.title = "Test"
	config.width = 1000
	config.height = 700
	val app = new LwjglApplication(this, config)
}

sealed abstract class BugDemonstration(val value: Int, val name: String) extends IntEnumEntry {

	import BugDemonstration._

	def create(skin: Skin): Actor

	def previous: BugDemonstration = if (this == values.head) this else BugDemonstration.withValue(this.value - 1)

	def next: BugDemonstration = if (this == values.last) this else BugDemonstration.withValue(this.value + 1)
}

case object BugDemonstration extends IntEnum[BugDemonstration] {

	val values: IndexedSeq[BugDemonstration] = findValues

	def withName(name: String): BugDemonstration = {
		values
			.find(_.name == name)
			.getOrElse(throw new IllegalArgumentException("name " + name + " not found"))
	}

	case object Example1 extends BugDemonstration(0, "HelloWorld") {
		override def create(skin: Skin): Actor = {
			val label = new Label("Hello World", skin)
			val table = new Table(skin)
			table.add(label)
			table
		}
	}

	case object Example2 extends BugDemonstration(1, "ToDo") {
		override def create(skin: Skin): Actor = {
			new Table(skin)
		}
	}
}
```
build.sbt
```

name := "Test 123"
version := "0.1"
scalaVersion := "2.12.1"
javacOptions ++= Seq("-source", "1.8", "-target", "1.8")
resolvers += Resolver.mavenLocal

libraryDependencies += "com.beachape" %% "enumeratum" % "1.5.2"

resolvers += "Sonatype OSS Snapshots" at "https://oss.sonatype.org/content/repositories/snapshots"

val gdxVersion = "1.9.5"

// libgdx
libraryDependencies += "com.badlogicgames.gdx" % "gdx" % gdxVersion
libraryDependencies += "com.badlogicgames.gdx" % "gdx-tools" % gdxVersion

// libgdx - pc
libraryDependencies += "com.badlogicgames.gdx" % "gdx-backend-lwjgl" % gdxVersion
libraryDependencies += "com.badlogicgames.gdx" % "gdx-platform" % gdxVersion classifier "natives-desktop"
```

**Comment by lloydmeta** [comment]
Does this still happen if you bumped the Enumeratum version to "1.5.7"?

**Comment by zeromancer** [comment]
oh, how have I missed, that i still have an old version ?
Updating to 1.5.7 has indeed resolved this issue.
Thanks

**Comment by lloydmeta** [comment]
Hey, no problem. FYI, the problem was with the macro and was fixed in #98 :)
